File: pe-gui/src/main/java/org/saar/gui/style/property/Coordinate.java
package org.saar.gui.style.property;

import org.saar.gui.style.IStyle;

public class Coordinate {

    private final IStyle parent;
    private final IStyle container;

    private int value;

    public Coordinate(IStyle parent, IStyle container) {
        this.parent = parent;
        this.container = container;
    }

    public void add(int value) {
        set(get() + value);
    }

    public void sub(int value) {
        set(get() - value);
    }

    public void set(Coordinate coordinate) {
        set(coordinate.get());
    }

    public int get() {
        return this.value;
    }

    public void set(int value) {
        this.value = value;
    }
}

File: pe-gui/src/main/kotlin/org/saar/gui/style/coordinate/Coordinate.kt
package org.saar.gui.style.coordinate

abstract class Coordinate(default: CoordinateValue = CoordinateValues.zero) : ReadonlyCoordinate {

    var value: CoordinateValue = default

    fun set(value: CoordinateValue) {
        this.value = value
    }

    fun set(pixels: Int) {
        this.value = CoordinateValues.pixels(pixels)
    }
}